dpos/store/dposstoredata.go

```diff
@@ -157,34 +304,46 @@ func (s *DposStore) persistBytesArray(batch Batch, bytesArray [][]byte, prefix D
 		}
 	}
 
-	batch.Put(key.Bytes(), value.Bytes())
-	return nil
+	return batch.Put(key, value.Bytes())
 }
 
-func (s *DposStore) persistDirectPeers(batch Batch, peers []*DirectPeers) error {
-	key := new(bytes.Buffer)
-	key.WriteByte(byte(DPOSDirectPeers))
+func (s *DposStore) getRewardData(height uint32,
+	prefix DataEntryPrefix) (*state.RewardData, error) {
 
-	value := new(bytes.Buffer)
+	key, err := s.getKey(height, prefix)
+	if err != nil {
+		return nil, err
+	}
+	data, err := s.db.Get(key)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := common.WriteVarUint(value, uint64(len(peers))); err != nil {
-		return err
+	rewards := &state.RewardData{
```

Comment:
Here can use a `NewRewardData` method instead of duplicate initializations.

Reply:
Fixed